hdf5/src/hl/references/standard.rs

```diff
@@ -0,0 +1,104 @@
+//! New standard reference types introduced in v1.12.0.
+//!
+//! These are gated on v1.12.1 since there appear to be multiple bugs in v1.12.0.
+//!
+use hdf5_sys::h5o::H5O_type_t;
+use hdf5_sys::h5r::H5R_type_t::H5R_OBJECT2;
+use hdf5_sys::h5r::{H5R_ref_t, H5Rcreate_object, H5Rdestroy, H5Rget_obj_type3, H5Ropen_object};
+
+use super::ObjectReference;
+use crate::internal_prelude::*;
+use crate::Location;
+
+/// A reference to a HDF5 item that can be stored in attributes or datasets.
+#[repr(transparent)]
+pub struct StdReference(H5R_ref_t);
```

Comment:
Move to `hdf5-types`

Reply:
Attempted this move but we need access to h5call! for the function in the drop implementation.